api4/group.go

```diff
@@ -403,6 +400,34 @@ func unlinkGroupSyncable(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
 
+func verifyLinkUnlinkPermission(c *Context, syncableType model.GroupSyncableType, syncableID string) {
+	switch syncableType {
+	case model.GroupSyncableTypeTeam:
+		if !c.App.SessionHasPermissionToTeam(c.App.Session, syncableID, model.PERMISSION_MANAGE_TEAM) {
+			c.SetPermissionError(model.PERMISSION_MANAGE_TEAM)
+			return
+		}
+	case model.GroupSyncableTypeChannel:
+		var permission *model.Permission
+
+		if channel, err := c.App.GetChannel(syncableID); err != nil {
```

Comment:
It can be done in a more idiomatic, moving the assignation outside the if and removing the else.